packages/experimental/Shimmer/src/Shimmer.types.ts

```diff
@@ -1,114 +1,156 @@
   /**
-   * Gradient tint color
-   * @defaultValue 'white'
+   * Border radius for the y-axis of a rounded rect.
+   * @defaultValue 0
    */
-  gradientTintColor?: ColorValue;
+  borderRadiusY?: number;
+
   /**
-   * Shimmer element tint color, no-op for image based shimmer
-   * @defaultValue 'E1E1E1' for light mode, '404040' for dark mode
+   * Note: x and y should be optional properties [or removed], with relative positioning being the default [or only] positioning mechanism.
+   * Note: The behavior of element intersections is undefined.
    */
-  shimmerTintColor?: ColorValue;
+
   /**
-   * Width of the shimmer view
-   * @defaultValue '200'
+   * The x-axis position of the rect element's top-left corner in the Shimmer relative to the origin [top-left, (0,0)].
+   * @defaultValue 0
    */
-  width?: number;
+  x: number;
+
   /**
-   * Height of the shimmer view
-   * @defaultValue '100'
+   * The y-axis position of the rect element's top-left corner in the Shimmer relative to the origin [top-left, (0,0)].
+   * @defaultValue 0
    */
-  height?: number;
+  y: number;
+}
+
+export interface ShimmerTokens extends IBackgroundColorTokens {
   /**
-   * Specifies the gradient angle, value should be anywhere between -1 and 1
-   * -1 to 1 maps to 90 degrees horizontally to ≈ 120 degrees
-   * @defaultValue '0'
+   * Specifies the Shimmer effect angle in degrees (produced by a gradient)
+   * @defaultValue 0
    */
   angle?: number;
+
+  /**
+   * Specifies the animation delay time in milliseconds
+   * @defaultValue 0
+   */
+  delay?: number;
+
+  /**
+   * Specifies the time required to traverse the control in milliseconds
+   *
+   * @defaultValue 2000
+   */
+  duration?: number;
+
   /**
-   * Specifies the gradient opacity, value should be less than 1
-   * @defaultValue '.7'
+   * Specifies the gradient opacity.
+   * @defaultValue 1
    */
   gradientOpacity?: number;
+
+  /**
+   * Color you see when the shimmer wave is not animating.
+   * @defaultValue theme.colors.bodyFrameDivider
+   */
+  shimmerColor?: ColorValue;
+
+  /**
+   * Defines the tip color of the wave which has a linear gradient.
+   * from shimmerColor to shimmerWaveColor to shimmerColor.
+   * @defaultValue '#E1E1E1'
+   */
+  shimmerWaveColor?: ColorValue;
+
+  /**
+   * Width of the Shimmer wave.
+   * @defaultValue '100%' of the 'width' property
+   */
+  shimmerWaveWidth?: number | string;
 }
 
 export interface ShimmerSlotProps extends ShimmerProps {
   root: SvgProps;
-  image: ImageProps;
 }
 
-export interface ShimmerProps extends ShimmerTokens {
-  /**
-   * Image to be used as a shimmer element
-   * @defaultValue 'null'
-   */
-  uri?: ImageURISource;
+export interface ShimmerProps extends ViewProps, ShimmerTokens {
   /**
-   * Array of ShimmerElement in a single view
+   * Shimmer shapes that define the masking effect of the Shimmer control.
    */
-  elements?: Array<ShimmerElement>;
+  elements?: Array<ShimmerCircleElement | ShimmerRectElement>;
 }
```

Comment:
Why did we remove the support for overlaying over an image? That's one of the scenarios we support in Native apple code with the FluentUI Apple Shimmer

Reply:
I referenced it in another comment but it's only implemented as a top line property/scenario in fluentUI iOS (if it's there for the native fluentui macos it isn't publicly documented).  I don't see the value in doing so when we're talking about adding alpha to colors and overlaying on top of an image.  We'd also be asserting our opinion about how that translucency should be applied to the gradient colors, which would need to be overridden for a different opinion.

It's also constraining in user control over the image relative to the shimmer e.g. aspect ratio; I don't believe we want to be in the business of managing that or extending our Shimmer API to expose those details.

Reply:
I'll do the followup in this thread. We don't have a FluentUI Apple macOS Shimmer, but that's mostly because we haven't needed one. I prototyped one a while back by porting the iOS code. 
The PR which added using a translucent shimmer in FluentUI Apple is here. I think it does a good job in explaining what the scenario was and why we added it. https://github.com/microsoft/fluentui-apple/pull/245

I'm not sure the decision making behind whether it belongs in FURN Shimmer or not, but it feels weird to me to take away that feature. I think for the purpose of expedience of this review, I'll leave it as a followup between you/Amanda/whoever has an opinion about what belongs and doesn't belong in the shimmer API. I'm not sure I have enough context to confirm.

Reply:
What I see being argued is that inverting the gradient opacity delta (shimmerColor-opacity >=< shimmerWaveColor-opacity) is better for an "image" shimmer element -- represented here with an icon or otherwise svg-compatible shape.  When you can customize the gradient colors, what's the difference?

The way the prop is now doesn't cover a lot -- You can only use one.  you can't make those shimmers shown because you don't have relative positioning at the moment -- if you check out the native code it cheats by walking the UI tree to manage giving a gradient to all the elements, which we can't do from JS. Or you do it through ImageProps crafted based on Shimmer implementation details; IMO an unkind coupling.  I'm making the assumption we're actually talking about raster images here (reason why is next paragraph), which will increase bundle sizes by theme colors * DPI, requiring coordinating resources w/ themes.  Warren Leung told me in another channel that using URI w/ SVG is considered a security hazard by Office [didn't get more info than that]

I'd rather add support for custom polygon/shape as an element type, unless the point is to offer color raster images which would be its own thing (not pictured, I haven't seen it before? and then I'd argue again not worth top-line support and maintenance).  This enables avoiding using resources/resource coordination/URI security concern, I imagine fits better/no worse than images/icons w/ the design pipeline, and can be made a top-line element type.